src/main/webapp/js/chart-loader.js

```diff
@@ -0,0 +1,31 @@
+
+google.charts.load("current", {packages: ["corechart"],});
+google.charts.setOnLoadCallback(
+  function() {
+    drawChartForDiffTypesOfTea();
```

Comment:
This function is no longer visible/in-scope. That is why I recommended doing the `google.charts.setOnLoadCallback` within the scope of the anonymous function in which the callback function is defined.

Reply:
I suggest reading these links to better understand what I'm requesting:

- [JavaScript Encapsulation With Anonymous Functions](https://www.codeproject.com/Articles/757949/JavaScript-Encapsulation-using-Anonymous-Functions)
- [What is the purpose in wrapping...?](https://stackoverflow.com/questions/2421911/what-is-the-purpose-of-wrapping-whole-javascript-files-in-anonymous-functions-li)

Note that in normal JavaScript development, *.js files aren't typically used directly; rather, a development *.js or *.ts file is compiled to a production *.js file that is actually used in serving in the web browser (see [Babel](https://babeljs.io/]), [Closure Compiler](https://developers.google.com/closure/compiler/), and TypeScript](https://www.typescriptlang.org/) for more info). This compilation process often automatically adds these anonymous function wrappers (`(function(){ ... })();`) to the files so that it does not need to be done manually. This compilation process will also typically [minify](https://en.wikipedia.org/wiki/Minification_(programming)) the JavaScript.

However, in our code base, we are not compiling the *.js files but rather using them directly. Since they are being used directly, it is useful to add these `(function(){ ... })();` wrappers manually. Although each *.html file currently uses its own, unique *.js file (and, therefore, it isn't at the moment super important), having this kind of encapsulation of the symbols of each file begins to become important when mixing *.js files together on multiple pages.

Reply:
Yup, this is fixed in the most recent commit : https://github.com/codeu-spring-2019-team-9/codeu-spring-2019-team-9/pull/13/commits/04e4fb13553802490af0d69715e7e2c6a18d0881
This is why I updated my commit to I'm dumb :P